File: source/cdk-stack/GenericDataCopyJob.scala
import com.amazonaws.services.glue.GlueContext
import com.amazonaws.services.glue.util.GlueArgParser
import com.amazonaws.services.glue.util.Job
import java.util.Calendar
import org.apache.spark.SparkContext
import org.apache.spark.sql.Dataset
import org.apache.spark.sql.Row
import org.apache.spark.sql.SaveMode
import org.apache.spark.sql.SparkSession
import org.apache.spark.sql.functions.from_json
import org.apache.spark.sql.streaming.Trigger
import scala.collection.JavaConverters._

object GenericDataCopyJob {
  def main(sysArgs: Array[String]): Unit = {
    val spark: SparkContext = new SparkContext()
    val glueContext: GlueContext = new GlueContext(spark)
    val sparkSession: SparkSession = glueContext.getSparkSession
    import sparkSession.implicits._

    // @params: [JOB_NAME, SECRET_ID, TABLE_DATA_PATH]
    val args = GlueArgParser.getResolvedOptions(sysArgs, Seq("JOB_NAME", "DRIVER_CLASS", "SECRET_ID", "TABLE_DATA_PATH").toArray)
    Job.init(args("JOB_NAME"), glueContext, args.asJava)
    val driverClass = args("DRIVER_CLASS")
    val secretId = args("SECRET_ID")
    val tableDataPath = args("TABLE_DATA_PATH")
    Class.forName(driverClass)

    val df = sparkSession.read.parquet(tableDataPath)
    val splits = tableDataPath.split('/')
    df.write
      .mode("overwrite")
      .format("jdbc")
      .option("url", secretId)
      .option("driver", driverClass)
      .option("dbtable", splits(splits.length - 2))
      .option("user", secretId)
      .save()
  }
}

File: source/cdk-stack/GenericDataCopyJob.py
import json
import sys

import boto3
import json
from awsglue.context import GlueContext
from awsglue.job import Job
from awsglue.transforms import *
from awsglue.utils import getResolvedOptions
from pyspark.context import SparkContext

secrets = boto3.client("secretsmanager")

engine_mapping = {
    "postgres": "postgresql"
}

## @params: [JOB_NAME]
args = getResolvedOptions(sys.argv, ["JOB_NAME", "SECRET_ID", "TABLE_DATA_PATH"])

sc = SparkContext()
glueContext = GlueContext(sc)
spark = glueContext.spark_session
job = Job(glueContext)
job.init(args['JOB_NAME'], args)

secret_id = args["SECRET_ID"]
table_data_path = args["TABLE_DATA_PATH"]

response = json.loads(secrets.get_secret_value(SecretId=secret_id)["SecretString"])

df = spark.read.parquet(table_data_path)
table_name = table_data_path.split('/')[-2]

engine = response["engine"]
if engine in engine_mapping:
    engine = engine_mapping[response["engine"]]

df.write \
    .mode("overwrite") \
    .format("jdbc") \
    .option("url", "jdbc:" + engine + "://" + response["host"] +
            ":" + str(response["port"]) + "/" + response["dbname"]) \
    .option("dbtable", table_name) \
    .option("user", response["username"]) \
    .option("password", response["password"]) \
    .save()

job.commit()
